### PR TITLE
added vmware and sql unmount functions

### PR DESCRIPTION
**Pull request by promiseowolabi**
# Description

Functionality to unmount VMware and SQL Live Mounts
## Related Issue

This project only accepts pull requests related to open issues.

https://github.com/rubrikinc/rubrik-sdk-for-python/issues/72

## Motivation and Context

Ongoing improvement of the Python SDK

## How Has This Been Tested?

Tested locally with demo labs

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTION](https://github.com/rubrikinc/rubrik-sdk-for-python/blob/master/CONTRIBUTING.md)** document.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rubrik_cdm/data_management.py [hunk 1]
```diff
@@ -990,3 +990,109 @@ def sql_live_mount(self, db_name, date, time, sql_instance=None, sql_host=None,
...
+            mount_id
+        except NameError:
+            raise InvalidParameterException("The mounted vSphere VM '{}' does exist, please check the name you provided.".format(
+                mounted_vm_name))
+        else:
+            #config = {}
+            #config['force'] = force
+
+            self.log(
+                "vsphere_live_mount: Unmounting the vSphere VM '{}'.".format(mounted_vm_name))
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: ` vsphere_live_mount: Unmounting the vSphere VM '{}'.".format(mounted_vm_name))`

`vsphere_live_mount` should be `vsphere_live_unmount`

File: rubrik_cdm/data_management.py [hunk 2]
```diff
@@ -990,3 +990,109 @@ def sql_live_mount(self, db_name, date, time, sql_instance=None, sql_host=None,
...
+        """Delete a Microsoft SQL Live Mount from the Rubrik cluster. 
+
+        Arguments:
+            mounted_db_name {str} -- The name of the Live Mounted database to be unmounted. 
+
+        Keyword Arguments:
+            sql_instance {str} -- The name of the MSSQL instance managing the Live Mounted database to be unmounted. 
+            sql_host {str} -- The name of the MSSQL host running the Live Mounted database to be unmounted. 
+            force {bool} -- Remove all data within the Rubrik cluster related to the Live Mount, even if the SQL Server database cannot be contacted. (default: {False})
+            timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {15})
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: `timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {15})` 

`(default: {15})` does not match the 30 second default in the function. 

Let's make sure the docs are correct here as well.

File: rubrik_cdm/data_management.py [hunk 3]
```diff
@@ -990,3 +990,109 @@ def sql_live_mount(self, db_name, date, time, sql_instance=None, sql_host=None,
...
+
+        Returns:
+            dict -- The full response of `DELETE /mssql/db/mount/{id}?force={bool}`.
+        """
+
+
+        if sql_instance is None or sql_host is None:
+                raise InvalidParameterException(
+                    "To live mount a mssql database the 'sql_instance' and 'sql_host' paramaters must be provided.")
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: `"To live mount a mssql database the 'sql_instance' and 'sql_host' paramaters must be provided."`

`live mount` should be something related to unmount

File: rubrik_cdm/data_management.py [hunk 4]
```diff
@@ -990,3 +990,109 @@ def sql_live_mount(self, db_name, date, time, sql_instance=None, sql_host=None,
...
+
+        mssql_host_id = self.object_id(sql_host, 'physical_host', timeout=timeout)
+        
+        self.log("sql_live_mount: Getting the list of instances on host {}.".format(sql_host))
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: `self.log("sql_live_mount: Getting the list of instances on host {}.".format(sql_host))`

`sql_live_mount` to `sql_live_unmount`

File: rubrik_cdm/data_management.py [hunk 5]
```diff
@@ -990,3 +990,109 @@ def sql_live_mount(self, db_name, date, time, sql_instance=None, sql_host=None,
...
+            'v1', '/mssql/instance?primary_cluster_id=local&root_id={}'.format(mssql_host_id), timeout)
+
+        for instance in mssql_instance['data']:
+            if instance['name'] == sql_instance:
+                sql_instance_id = instance['id']
+                break
+        else:
+            raise InvalidParameterException("The SQL instance {} does not exist, please provide a valid instance".format(sql_instance))
+
+        self.log("sql_live_mount: Getting the list of databases on the instance {}, on host {}.".format(sql_instance, sql_host))
```
Comment (drew-russell) [inline rubrik_cdm/data_management.py]: `sql_live_mount` to `sql_live_unmount`

File: rubrik_cdm/data_management.py [hunk 6]
```diff
@@ -990,3 +990,109 @@ def sql_live_mount(self, db_name, date, time, sql_instance=None, sql_host=None,
...
+
+    def vsphere_live_unmount(self, mounted_vm_name, force=False, timeout=30):  # pylint: ignore
+        """Delete a vSphere Live Mount from the Rubrik cluster. 
+
+        Arguments:
+            mounted_vm_name {str} -- The name of the Live Mounted vSphere VM to be unmounted. 
+
+        Keyword Arguments:
+            force {bool} -- Force unmount to remove metadata when the datastore of the Live Mount virtual machine was moved off of the Rubrik cluster. (default: {False})
+            timeout {int} -- The number of seconds to wait to establish a connection the Rubrik cluster before returning a timeout error. (default: {15})
```
[the remark this answers is not in the document]
Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: Corrected, thanks

File: rubrik_cdm/data_management.py [hunk 7]
```diff
@@ -990,3 +990,109 @@ def sql_live_mount(self, db_name, date, time, sql_instance=None, sql_host=None,
...
+        else:
+            raise InvalidParameterException("The mounted VM '{}' does not exist, please provide a valid instance".format(mounted_vm_name))
+
+        try:
+            mount_id
+        except NameError:
+            raise InvalidParameterException("The mounted vSphere VM '{}' does exist, please check the name you provided.".format(
+                mounted_vm_name))
+        else:
+            #config = {}
```
[the remark this answers is not in the document]
Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: Comments removed, thanks

File: rubrik_cdm/data_management.py (hunk 1 above)
Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: Changes made thanks

File: rubrik_cdm/data_management.py (hunk 2 above)
Comment (promiseowolabi) [reply rubrik_cdm/data_management.py]: Changes made, thanks

**Pull request merged**